### PR TITLE
Remove `vendor` directory

### PR DESCRIPTION
**Pull request by rishabh-11**
**What this PR does / why we need it**:
This PR gets rid of vendor directory in favour of go modules

**Which issue(s) this PR fixes**:
Fixes part of #902

**Special notes for your reviewer**:

**Release note**:

```other operator
Removed vendor directory
```

### CODE REVIEW

File: .reuse/dep5 [hunk 1]
```diff
@@ -67,290 +67,6 @@ Copyright: gRPC authors
...
-Copyright: 2024 github.com/blang/semver/v4 contributors.
-License: MIT
-
-Files: vendor/gopkg.in/inf.v0/*
-Copyright: 2024 gopkg.in/inf.v0 contributors.
-License: BSD-3-Clause
\ No newline at end of file
+Files:
+Copyright:
+License:
```
Comment (RaphaelVogel) [inline .reuse/dep5:72]: those line should be commented out

**Review by RaphaelVogel — CHANGES_REQUESTED** [review]

**Review by RaphaelVogel — APPROVED** [review]

**Comment by RaphaelVogel** [comment]
from compliance perspective it looks good now, have not validated vendor removal itself

**Review by elankath — CHANGES_REQUESTED** [review]
The script `hack/generate-code` is still leveraging vendor folder. Is this intended ?

**Comment by rishabh-11** [comment]
No, need to fix that. Thanks

**Comment by aaronfern** [comment]
Please remove reference of vendor in [`hack/add_license_headers`](https://github.com/gardener/machine-controller-manager/blob/master/hack/add_license_headers.sh#L23) as well

**Review by elankath — APPROVED** [review]
/lgtm

**Pull request merged**